worker/uniter/runner/context/context.go

```diff
@@ -783,6 +784,15 @@ func (ctx *HookContext) Flush(process string, ctxErr error) (err error) {
 		defer ctx.handleReboot(&err)
 	}
 
+	// When processing config changed hooks we need to ensure that the
+	// relation settings for the unit endpoints are up to date after
+	// potential changes to already bound endpoints.
+	if process == string(hooks.ConfigChanged) {
```

Comment:
Note: while `Flush` seems to expect a _process_ name, this is actually the hook name! `Flush` gets invoked by `runCharmHookWithLocation` which passes the hook name so this will also work on windows.